x/wasm/internal/keeper/query_plugins.go

```diff
@@ -146,6 +164,71 @@ func NoCustomQuerier(sdk.Context, json.RawMessage) ([]byte, error) {
 	return nil, wasmvmtypes.UnsupportedRequest{Kind: "custom"}
 }
 
+func IBCQuerier(wasm *Keeper, channelKeeper types.ChannelKeeper) func(ctx sdk.Context, caller sdk.AccAddress, request *wasmvmtypes.IBCQuery) ([]byte, error) {
+	return func(ctx sdk.Context, caller sdk.AccAddress, request *wasmvmtypes.IBCQuery) ([]byte, error) {
+		if request.PortID != nil {
+			contractInfo := wasm.GetContractInfo(ctx, caller)
+			res := wasmvmtypes.PortIDResponse{
+				PortID: contractInfo.IBCPortID,
+			}
+			return json.Marshal(res)
+		}
+		// I just discovered we cannot query by port (how odd since they store it like this)
+		// Anyway, I will allow Port as an optional filter
+		// TODO: let's revisit this query (if needed, if we should change it)
+		if request.ListChannels != nil {
+			portID := request.ListChannels.PortID
+			var channels wasmvmtypes.IBCEndpoints
+			channelKeeper.IterateChannels(ctx, func(ch types.IdentifiedChannel) bool {
+				if portID == "" || portID == ch.PortId {
+					newChan := wasmvmtypes.IBCEndpoint{
+						PortID:    ch.PortId,
+						ChannelID: ch.ChannelId,
+					}
+					channels = append(channels, newChan)
+				}
+				return false
+			})
+			res := wasmvmtypes.ListChannelsResponse{
+				Channels: channels,
+			}
+			return json.Marshal(res)
+		}
+		if request.Channel != nil {
+			channelID := request.Channel.ChannelID
+			_ = channelID
```

Comment:
nit: `_` unused

Reply:
Good point, left from a WIP version so it would compile